roles/cluster_deploy_catalog_source/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Fail if catalog_image is undefined
+  fail:
+    msg: Must provide a catalog_image
+  when: catalog_image is undefined
+
+- name: Fail if catalog_registry is undefined
+  fail:
+    msg: Must provide a catalog_registry
+  when: catalog_registry is undefined
+
+- name: Set default catalog_version (N-1) if undefined
+  set_fact:
+    catalog_version: "v{{ openshift_release_major }}.{{ openshift_release_minor|int - 1 }}"
+  when: catalog_version is undefined
+
+- name: Set catalog_name
+  set_fact:
+    catalog_name: "{{ catalog_image }}-cs"
+
+
+- name: 'Deploy {{ catalog_image }}:{{ catalog_version }} CatalogSource'
+  block:
+  - name: Catalog source version
+    debug:
+      msg: "Current OCP version: {{ openshift_release }}, Setting catalog tag {{ catalog_version }}"
+  
+  - name: Template CatalogSource for previous OCP release
+    template:
+      src: "{{ operator_catalog_previous }}"
+      dest: "{{ artifact_extra_logs_dir }}/{{ catalog_image }}-catalog-previous.yml"
+      mode: 0400
+  
+  - name: Apply CatalogSource for previous OCP release
+    command: oc apply -f "{{ artifact_extra_logs_dir }}/{{ catalog_image }}-catalog-previous.yml"
+    changed_when: true
+
+  - name: Verify catalog pod is ready
+    shell: 
+      cmd: |
+        oc get pods -n openshift-marketplace -ojson \
+        | jq -r '.items[]|.metadata.name+" "+.status.phase|select(match("{{ catalog_name }}"))' \
+        | awk '{print $2}' | grep Running
+    retries: 25
+    delay: 30
+    register: catalog_pod_ready
+    until: catalog_pod_ready.rc == 0
+
+  - name: Verify Catalog source
+    shell:
+      cmd: oc get catalogsource -n openshift-marketplace -ojson | jq -r '.items[]|.metadata.name' | grep {{ catalog_name }}
```

Comment:
hum, why writting this as 
```
    shell:
      cmd: <command>
```
instead of
```
    shell: <command>
```
?

Reply:
My ansible linter liked it better. 🤷🏽‍♂️

Reply:
I would mute this one ... or you'll have to change it `152` times to align the rest of the repo 🙃